src/kvm.c

```diff
@@ -351,14 +356,28 @@ kvm_ioctl_run_attach_auxstr(struct tcb *const tcp,
 	tcp->auxstr = xlookup(kvm_exit_reason, vcpu_run_struct.exit_reason);
 	if (!tcp->auxstr)
 		tcp->auxstr = "KVM_EXIT_???";
+	if (decode_kvm_run_structure > DECODE_KVM_RUN_STRUCTURE_EXIT_REASON)
+		tcp->vcpu_leaving = &vcpu_run_struct;
 }
 
 static int
 kvm_ioctl_decode_run(struct tcb *const tcp)
 {
 
-	if (entering(tcp))
+	if (entering(tcp)) {
+		if (decode_kvm_run_structure > DECODE_KVM_RUN_STRUCTURE_EXIT_REASON) {
+			int fd = tcp->u_arg[0];
+			struct vcpu_info *info = vcpu_get_info(tcp, fd);
+			if (info) {
+				if (!tcp->vcpu_entering)
+					tcp->vcpu_entering
+						= xzalloc(sizeof(*tcp->vcpu_entering));
+				kvm_ioctl_run_umove(tcp, info,
+						    tcp->vcpu_entering);
```

Comment:
What if kvm_ioctl_run_umove failed?  Wouldn't vcpu_entering contain garbage in that case?